sql/api/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -331,7 +331,7 @@ object ScalaReflection extends ScalaReflection {
       case t if isSubtype(t, localTypeOf[java.time.Instant]) => STRICT_INSTANT_ENCODER
       case t if isSubtype(t, localTypeOf[java.time.LocalDateTime]) => LocalDateTimeEncoder
       case t if isSubtype(t, localTypeOf[VariantVal]) => VariantEncoder
-      case t if isSubtype(t, localTypeOf[Row]) => UnboundRowEncoder
+      case t if isRowEncoderSupported && isSubtype(t, localTypeOf[Row]) => UnboundRowEncoder
```

Comment:
I am curious to find out what will break :)....

Can you also check the call sites that use this flag?